junipervpn/vpn.py

```diff
@@ -0,0 +1,395 @@
+            stdin = stdin.replace('%HOST%', args.host)
+            p.communicate(input=stdin.encode('ascii'))
+        else:
+            ret = p.wait()
+        ret = p.returncode
+        # Reset child to None so we don't try to kill a completed
+        # process:
+        self.child = None
+
+        # Openconnect specific
+        if ret == 2:
+            self.cj.clear(args.host, '/', 'DSID')
+            self.r = self.br.open(self.r.geturl())
+
+    def stop(self):
+        if self.child:
+            print("Interrupt received, ending external program...")
+            # Use SIGINT due to openconnect behavior where SIGINT will
+            # run the vpnc-compatible script to clean up changes but
+            # not upon SIGTERM.
+            # http://permalink.gmane.org/gmane.network.vpn.openconnect.devel/2451
+            try:
+                self.child.send_signal(signal.SIGINT)
+                self.child.wait()
+            except OSError:
+                pass
+        sys.exit(0)
+
+def main():
+    parser = argparse.ArgumentParser(conflict_handler='resolve')
+    parser.add_argument('-h', '--host', type=str,
+                        help='VPN host name')
+    parser.add_argument('-r', '--realm', type=str,
+                        help='VPN realm')
+    parser.add_argument('-u', '--username', type=str,
+                        help='User name')
+    parser.add_argument('-p', '--pass_prefix', type=str,
+                        help="Secondary password prefix")
+    parser.add_argument('-o', '--oath', type=str,
+                        help='OATH key for two factor authentication (hex)')
+    parser.add_argument('-c', '--config', type=str,
+                        help='Config file')
+    parser.add_argument('-s', '--stdin', type=str,
+                        help="String to pass to action's stdin")
+    parser.add_argument('-d', '--device-id', type=str,
+                        help="Hex device ID")
+    parser.add_argument('-f', '--enable-funk', action='store_true',
+                        help="Request funk message")
+    parser.add_argument('-H', '--hostname', type=str,
+                        help="Hostname to pass with funk request")
+    parser.add_argument('-p', '--platform', type=str,
+                        help="Platform type to pass with funk request")
+    parser.add_argument('-a', '--hwaddr', type=str,
+                        help="Comma separated list of hwaddrs to pass with funk request")
+    parser.add_argument('-C', '--certs', type=str,
+                        help="Comma separated list of pem formatted certificates for funk response")
+    parser.add_argument('-U', '--user-agent', type=str,
+                        help="User agent string")
+    parser.add_argument('action', nargs=argparse.REMAINDER,
+                        metavar='<action> [<args...>]',
+                        help='External command')
+
+    args = parser.parse_args()
+    args.password = None
```

Comment:
Apart from a removed import __future__, changing `args.__dict__[...]` to `args.password` seems to be the only non-required change added during the rename of this file.

It's not super important, but maybe worth reverting in the interests of accurate history? (i.e. to keep this as close to a straight rename as possible). It could then be improved in the argparse PR.